docs/proposals/fluentd-limit-requests-to-apiserver.md

```diff
@@ -0,0 +1,63 @@
+# Modifications to Fluentd Metadata Retrieval Proposal
+
+##Motivation
+In the current design (v3.4) of the EFK stack, there is a Fluentd pod deployed to each Openshift node. Fluentd is responsible for reading container logs, adding pod metadata (e.g. labels, namespace) to individual log entries, and sending the logs to the Elasticsearch cluster. The pod metadata is sourced by watching the Openshift API server for Pod information via the [Kubernetes Metadata Filter](https://github.com/fabric8io/fluent-plugin-kubernetes_metadata_filter)[2]. Testing at larger scales of Openshift clusters has shown issues where Fluentd is unable to connect to the API server because it is limited by the **maxRequestsInFlight** configuration setting [1].  This results in logs not being shipped to the aggregator and thus being unavailable.
```

Comment:
API server connections (watches) are kept open to keep metadata that is only available currently from the API server (labels, annotations, etc) up to date & attached to log events as they happen. This is cached to reduce the number of API server calls, but as it's a watch it keeps a long lived connection alive to keep the metadata up to date.

If you wanted to remove the persistent connection, you could disable watch & set a reasonable TTL on the cache so that cached data is expired regularly. This will increase the number of short lived connections to the API server significantly though. Also this means the metadata should be considered stale between API calls.